nixos/modules/services/network-filesystems/davfs2.nix

```diff
@@ -1,61 +1,70 @@
 { config, lib, pkgs, ... }:
 
-with lib;
-
 let
-  toStr = value:
-    if true == value then "yes"
-    else if false == value then "no"
-    else toString value;
+  inherit (lib.attrsets) optionalAttrs;
+  inherit (lib.generators) toINIWithGlobalSection;
+  inherit (lib.lists) optional;
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) literalExpression mkEnableOption mkOption;
+  inherit (lib.strings) escape;
+  inherit (lib.types) attrsOf bool either int lines oneOf str submodule;
 
   cfg = config.services.davfs2;
-  format = pkgs.formats.toml { };
-  configFile = let
-    settings = mapAttrsToList (n: v: "${n} = ${toStr v}") cfg.settings;
-  in pkgs.writeText "davfs2.conf" ''
-    ${concatStringsSep "\n" settings}
+
+  escapeString = escape ["\"" "'"];
+
+  formatValue = value:
+    if true == value then "1"
+    else if false == value then "0"
+    else if builtins.isString value then "\"${escapeString value}\""
+    else toString value;
+
+  configFile = pkgs.writeText "davfs2.conf" ''
+    ${toINIWithGlobalSection {
+      mkSectionName = escapeString;
+      mkKeyValue = k: v: "${k} ${formatValue v}";
+    } cfg.settings}
     ${cfg.extraConfig}
```

Comment:
The one question I have is how to handle this - it's wrong to simply append, as the config can end up in the wrong section. Should we simply remove this option?

Reply:
On second thought, I'll keep it deprecated, but enforce using one or the other (`settings` or `extraConfig`).